### PR TITLE
Cloudflare R2 Integration 

### PR DESCRIPTION
**Pull request by sarahwooders**
- Update `skyplane init` to setup Cloudflare R2 access keys (stored in `SkyplaneConfig`) 
- Planner: One-sided transfers (create only VMs in source/destination regions) 
- Modify places where `region_tag` of destinations is mapped to instance regions, since with one-sided transfers the destination region may not correspond to the sink instance region

Todo 

- [ ] Test with all cloud providers 
- [x] Fix access keys issues

### CODE REVIEW

File: skyplane/cli/cli_init.py [hunk 1]
```diff
@@ -63,6 +63,16 @@ def load_aws_config(config: SkyplaneConfig, non_interactive: bool = False) -> Sk
         return config
 
 
+def load_cloudflare_config(config: SkyplaneConfig, non_interactive: bool = False) -> SkyplaneConfig:
+    if typer.confirm("    Do you want to configure Cloudflare support in Skyplane?", default=True):
```
Comment (abiswal2001) [inline skyplane/cli/cli_init.py:67]: `noninteractive or typer.confirm`

File: skyplane/api/tracker.py [hunk 2]
```diff
@@ -264,10 +264,11 @@ def monitor_single_dst_helper(dst_region):
...
         """Monitor the tranfer by copying remote gateway logs and show transfer stats by hooks"""
         # todo implement transfer monitoring to update job_complete_chunk_ids and job_pending_chunk_ids while the transfer is in progress
-        region_sinks = self.dataplane.topology.sink_instances()
-        sinks = region_sinks[region_tag]
-        # for region_tag, sink_gateways in self.dataplane.topology.sink_gateways().items():
-        # sink_regions = set([sink.region for sink in sinks])
+
+        # regions that are sinks for specific region tag
+        # TODO: should eventualy map bucket to list of instances
+        sinks = [n for nodes in self.dataplane.topology.sink_instances(region_tag).values() for n in nodes]
```
Comment (abiswal2001) [inline skyplane/api/tracker.py:270]: I'm a bit confused as to what this is doing?

File: skyplane/api/transfer_job.py [hunk 3]
```diff
@@ -56,7 +57,9 @@ def __init__(self, src_obj: ObjectStoreObject, dst_objs: Dict[str, ObjectStoreOb
 class GatewayMessage:
     def __init__(self, chunk: Optional[Chunk] = None, upload_id_mapping: Optional[Dict[str, Tuple[str, str]]] = None):
         self.chunk = chunk
-        self.upload_id_mapping = upload_id_mapping
+        self.upload_id_mapping = (
+            upload_id_mapping  # region_tag: (upload_id, dst_key) # TODO: map both bucket + region to enable multiple buckets per region
+        )
```
Comment (abiswal2001) [inline skyplane/api/transfer_job.py:62]: could you clarify this mapping change?

File: skyplane/utils/path.py [hunk 4]
```diff
@@ -16,7 +16,14 @@ def is_plausible_local_path(path_test: str):
             return True
         return False
 
-    if path.startswith("cos://"):
+    if (path.startswith("https://") or path.startswith("http://")) and "r2.cloudflarestorage.com" in path:
+        regex = re.compile(r"https?://([^/]+).r2.cloudflarestorage.com/([^/]+)/?(.*)")
+        match = regex.match(path)
+        if match is None:
+            raise ValueError(f"Invalid Azure path: {path}")
```
Comment (abiswal2001) [inline skyplane/utils/path.py:23]: "Invalid Cloudflare path"

File: skyplane/api/tracker.py (hunk 2 above)
Comment (sarahwooders) [reply skyplane/api/tracker.py:270]: This is grabbing all the instances that are writing to a bucket in `region_tag` - previously the sink instances were defined as the instances in the sink region, but that doesn't hold for one-sided transfers

File: skyplane/api/transfer_job.py (hunk 3 above)
Comment (sarahwooders) [reply skyplane/api/transfer_job.py:62]: We eventually want to support uploading to multiple buckets in the same region, so each bucket in the same region would correspond to different upload Ids - so we'd need to change this part of the code. I can update the comment.

**Pull request merged**